dotCMS/src/main/java/com/dotcms/graphql/business/GraphqlAPIImpl.java

```diff
@@ -160,137 +97,37 @@ private void printSchema() {
-
-        fields.forEach((field)->{
-            // skip field.variable not sticking to the regex
-            if(!field.variable().matches(TYPES_AND_FIELDS_VALID_NAME_REGEX)
-                || field instanceof RelationshipsTabField) {
-                return;
-            }
+    @LogTime(loggingLevel = "INFO")
+    private GraphQLSchema generateSchema() throws DotDataException {
+        final Set<GraphQLType> graphQLTypes = new HashSet<>();
 
-            if(!(field instanceof RowField) && !(field instanceof ColumnField)) {
-                if (field instanceof RelationshipField) {
-                    try {
-                        handleRelationshipField(contentType, builder, field, graphqlObjectTypes);
-                    } catch(DotStateException e) {
-                        Logger.error(this, "Unable to create relationship field", e);
-                    }
-                } else {
-                    builder.field(newFieldDefinition()
-                        .name(field.variable())
-                        .type(field.required()
-                            ? nonNull(getGraphqlTypeForFieldClass(field.type(), field))
-                            : getGraphqlTypeForFieldClass(field.type(), field))
-                        .dataFetcher(getGraphqlDataFetcherForFieldClass(field.type()))
-                    );
-                }
+        for (GraphQLTypesProvider typesProvider : typesProviders) {
+            try {
+                graphQLTypes.addAll(typesProvider.getTypes());
+            } catch (DotDataException e) {
+                Logger.error("Unable to get types for type provider:" + typesProvider
+                        .getClass(), e);
             }
-        });
-
-        builder.withInterface(InterfaceType.CONTENTLET.getType());
-        final GraphQLObjectType graphQLType = builder.build();
-
-        graphqlObjectTypes.put(graphQLType.getName(), graphQLType);
-    }
-
-    private void handleRelationshipField(final ContentType contentType, GraphQLObjectType.Builder builder,
-                                         final Field field, final Map<String, GraphQLObjectType> typesMap) {
-
-        final ContentType relatedContentType;
-        try {
-            relatedContentType = getRelatedContentTypeForField(field, APILocator.systemUser());
-        } catch (DotSecurityException | DotDataException e) {
-            throw new DotStateException("Unable to create relationship field type for field: " + contentType.variable() + "." + field.variable(), e);
         }
 
-        Relationship relationship;
+        List<GraphQLFieldDefinition> fieldDefinitions = new ArrayList<>();
 
-        try {
-            relationship = relationshipAPI.getRelationshipFromField(field,
-                APILocator.systemUser());
-        } catch (DotDataException | DotSecurityException e) {
-            throw new DotRuntimeException(e);
+        for (GraphQLFieldsProvider fieldsProvider : fieldsProviders) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'fieldsProvider' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)